src/objects/entities/Entity.ts

```diff
@@ -0,0 +1,205 @@
+import Phaser from 'phaser';
+import keepUprightStratergies from '@/objects/Enums/Physics';
+import keepUpright from '@/helpers/keepUprightStratergy';
+import { Vector } from 'matter';
+
+type physicsConfigType = {
+  shape: Phaser.Types.Physics.Matter.MatterSetBodyConfig,
+  chamfer?: Phaser.Types.Physics.Matter.MatterChamferConfig
+  frictionAir?: number,
+  bounce?: number,
+};
+
+type animationsConfigType = {
+  animationKey: string, 
+  start: number, 
+  end: number, 
+  fps: number,
+  repeat?: number | undefined
+};
```

Comment:
i think we can pull these straight out of Phaser package

Reply:
If we use Phasers built in Animation type it doesn't seem to have an endFrame param and requires we pass it an AnimationFrame object which is built from the function `anims.generateFrameNumbers()`. Might be easier to keep our custom type here or just use a custom type for the Frames. `end ` is only needed for `generateFrameNumbers` 

https://newdocs.phaser.io/docs/3.60.0/Phaser.Types.Animations